### DISCUSSION TITLE
Unable to Fix Incorrect Address

### DISCUSSION THREAD

**Discussion by AmyKinKC** [code4recovery/12-step-meeting-list#1405] [General]
Hello!
I hope you can help me. Any insight to a fix for this would be greatly appreciated.

I am trying to change the address for this meeting, but it will not update to the correct address.
Correct Address is 1000 East State Route 92, Kearney, MO 64060

Here is the link to the meeting.
https://wamo-aa.org/meetings/6730/?tsml-day=any&tsml-query=Kearney%20Group

When I enter East and click out of the address line it reverts back to 1000 State Route 92, Kearney, MO 64060.

**Comment by joshreisner** [comment]
hi @AmyKinKC sorry about that. sometimes google's geocoding service makes mistakes, even when you can find the same address correctly on google maps

looks like this is the correct location: https://www.google.com/maps/place/First+United+Methodist+Church/@39.3686208,-94.3413572,17z/data=!4m6!3m5!1s0x87c1adb417146381:0xf1a76517c313882f!8m2!3d39.3687825!4d-94.3399946!16s%2Fg%2F1tkrm9yd?entry=ttu

i think you can fix this locally by overriding it by adding this code to your theme's functions.php

```php
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '1000 East State Rte 92, Kearney, MO 64060, USA' => array(
            'formatted_address' => '1000 East State Rte 92, Kearney, MO 64060, USA',
            'city' => 'Kearney',
            'latitude' => 39.369751,
            'longitude' => -94.339668,
            'approximate' => 'no',
        ),
    ));
}
```

once you've done that, using the exact address `1000 East State Rte 92, Kearney, MO 64060, USA` should go to the right place

**Reply by AmyKinKC** [reply]
Hi Josh. Thank You for getting back to me. I tried to figure out where to paste the code you gave me. But I am not sure where it goes or how to insert it. I am not a coder. The screenshot is where I think it needs to go... 

![kearney group](https://github.com/code4recovery/12-step-meeting-list/assets/99058201/97f394dc-134c-48e7-acff-ddd31c2b5387)

**Reply by joshreisner** [reply]
hi @AmyKinKC you want to add this to your theme's `functions.php` file. 

the recommended way to do this is through your hosting file manager. perhaps @CheriJones can help you?

**Comment by AmyKinKC** [comment]
Hi @joshreisner Thank You for your help. I think I find someone to help place the code. Cheri is no longer a part of our Tech Team.

**Reply by CheriJones** [reply]
thanks @joshreisner i applied the fix and it works perfectly

**Comment by Tech4TriCounty** [comment]
Didn't want to open another thread, if you'd like I can. I have a similar issue. Google returns my addresses occasionally with a "o" or an "o o".

Here is what it looks like: "207 E Pitman St o O'Fallon, MO 63366". Any update I try will not stick. Can I use this method similarly? We have several at this club. We also use the PDF Generator and it contains those same address bugaboos see attached image. 

https://tricountyaa.org/meetings/212-club-saturdays-at-600-pm/
feed data:
{"id":4741,"name":"Group 979","slug":"group-979","updated":"2023-02-09 16:30:48","location_id":12893,"url":"https:\/\/tricountyaa.org\/meetings\/group-979\/","day":1,"time":"07:00","end_time":"08:00","time_formatted":"7:00 am","edit_url":"https:\/\/tricountyaa.org\/wp-admin\/post.php?post=4741&action=edit","types":["DR","O"],"author":"tricountyaa_admin","location":"212 Club","location_notes":"lower level rear.","location_url":"https:\/\/tricountyaa.org\/locations\/212-club\/","formatted_address":"207 E Pitman St o, O'Fallon, MO 63366, USA","approximate":"no","latitude":38.8069804,"longitude":-90.6976733,"region_id":358,"region":"O'Fallon","regions":["O'Fallon"],"group_id":4740,"group":"Group 979","district":"District 19","district_id":357,"attendance_option":"in_person"}

<img width="289" alt="Screenshot 2024-05-16 at 10 27 38 AM" src="https://github.com/code4recovery/12-step-meeting-list/assets/132401769/34763e75-db78-44b4-b3b8-4c23141359cb">

**Reply by joshreisner** [reply]
hi @Tech4TriCounty when i geocode this address i get `207 E Pitman St, O'Fallon, MO 63366, USA` back, which looks correct. i think it might have cached a bad result from google at one point. we don't have a good way of flushing the cache, so the php approach is the best option, unless you're comfortable running database commands

```
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '207 E Pitman St, O'Fallon, MO 63366, USA' => array(
            'formatted_address' => '207 E Pitman St, O'Fallon, MO 63366, USA',
            'city' => 'O\'Fallon',
            'latitude' => 38.8069804,
            'longitude' => -90.6976733,
            'approximate' => 'no',
        ),
    ));
}
```

re the screenshot, that doesn't look like one of our products - not sure how you'd edit that. is it this plugin? https://wordpress.org/plugins/12-step-meeting-pdf-generator/

**Reply by Tech4TriCounty** [reply]
Thank you Josh,
I am comfortable running the database commands to clean up the db? I can get command line access to the mysql db.

Yes that is the plugin. I got it almost at simultaneously with the TSML plug in and I did not see any other indicator than it was part of the TSML plugin itself. Does C4R offer such a tool? I will ask the developer of that plugin for information on editing the file. Without a replacement piece, it would be my only direction.

I really appreciate your help

**Reply by joshreisner** [reply]
we have a different PDF solution which you can access on the Import / Export page. it has different features though, and doesn't support the tri-fold format which you might prefer

the sql command is `DELETE FROM wp_options WHERE option_name = 'tsml_addresses'`

this assumes the default table prefix of `wp_`, but your site may have a different prefix, depending on how your wordpress is set up

**Reply by Tech4TriCounty** [reply]
I tried editing and dropping the database and the data just keeps coming back. Looks like the only option is the function above. 

How does this code work, where does tsml_custom_address get set?
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '207 E Pitman St, O'Fallon, MO 63366, USA' => array(
            'formatted_address' => '207 E Pitman St, O'Fallon, MO 63366, USA',
            'city' => 'O\'Fallon',
            'latitude' => 38.8069804,
            'longitude' => -90.6976733,
            'approximate' => 'no',
        ),
    ));
}

When I edit the theme's function.php it breaks the site. Where should I be putting this, in the function page?

page starts with: <?php

define( 'SITEORIGIN_THEME_VERSION', '1.10.2' );
define( 'SITEORIGIN_THEME_JS_PREFIX', '.min' );

include get_template_directory() . '/inc/settings/settings.php';
include get_template_directory() . '/inc/customizer/customizer.php';

include get_template_directory() . '/inc/widgets.php';
include get_template_directory() . '/inc/settings.php';
include get_template_directory() . '/inc/customizer.php';
include get_template_directory() . '/inc/gallery.php';
include get_template_directory() . '/inc/panels.php';
include get_template_directory() . '/inc/recommended-plugins.php';
include get_template_directory() . '/inc/legacy.php';

**Reply by Tech4TriCounty** [reply]
Not sure what to say. It cleared up in the database and on the meeting pages, but in the admin under certain conditions it loads the address correctly and then flash loads the address with a superfluous character. I have to be careful not to update it that way. I also was able to edit the code after finding it in the plugins/code folder. I appreciate your time and effort here. If you want any help with clearing up the issue or it has legs beyond me, I am more than willing to see it through. I consult with larger CS entities and your PDF generator may be useful for those that don't use the trifold. We use the trifold as simple AA literature with the 20 questions etc. We only have 130 meetings in our tricounty area, but our neighboring St. Louis has 700+ meetings and a trifold wouldn't work for them.

**Reply by joshreisner** [reply]
we are working on a new solution that hopefully will make this process of confirming / fixing addresses easier and more transparent going forward 🤞 

if it's flashing the wrong data at you, maybe one option is to add another entry (make the whole thing look like this):

```php
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '207 E Pitman St, O'Fallon, MO 63366, USA' => array(
            'formatted_address' => '207 E Pitman St, O'Fallon, MO 63366, USA',
            'city' => 'O\'Fallon',
            'latitude' => 38.8069804,
            'longitude' => -90.6976733,
            'approximate' => 'no',
        ),
        '207 E Pitman St o, O'Fallon, MO 63366, USA' => array(
            'formatted_address' => '207 E Pitman St, O'Fallon, MO 63366, USA',
            'city' => 'O\'Fallon',
            'latitude' => 38.8069804,
            'longitude' => -90.6976733,
            'approximate' => 'no',
        ),
    ));
}
```

**Comment by Tech4TriCounty** [comment]
Oops we would also like to remove the MO as all our meetings are in missouri. I am not afraid to tear into php files if I need to. 
<img width="289" alt="Screenshot 2" src="https://github.com/code4recovery/12-step-meeting-list/assets/132401769/80275fae-966a-485e-a0ca-2cc9faeecc5d">
